x-pack/plugins/security_solution/public/detections/components/rules/eql_query_bar/footer.tsx

```diff
@@ -28,22 +49,169 @@ const FlexGroup = styled(EuiFlexGroup)`
+  const closeEqlSettingsHandle = useCallback(() => setIsOpenEqlSettings(false), []);
+  const handleEventCategoryField = useCallback(
+    (opt: EuiComboBoxOptionOption[]) => {
+      if (onOptionsChange) {
+        if (opt.length > 0) {
+          onOptionsChange('eventCategoryField', opt[0].label);
+        } else {
+          onOptionsChange('eventCategoryField', null);
+        }
+      }
+    },
+    [onOptionsChange]
+  );
+  const handleTiebreakerField = useCallback(
+    (opt: EuiComboBoxOptionOption[]) => {
+      if (onOptionsChange) {
+        if (opt.length > 0) {
+          onOptionsChange('tiebreakerField', opt[0].label);
+        } else {
+          onOptionsChange('tiebreakerField', null);
+        }
+      }
+    },
+    [onOptionsChange]
+  );
+  const handleTimestampField = useCallback(
+    (opt: EuiComboBoxOptionOption[]) => {
+      if (onOptionsChange) {
+        if (opt.length > 0) {
+          onOptionsChange('timestampField', opt[0].label);
+        } else {
+          onOptionsChange('timestampField', null);
+        }
+      }
+    },
+    [onOptionsChange]
+  );
+
+  const eventCategoryField = useMemo(
+    () =>
+      optionsSelected?.eventCategoryField != null
+        ? [{ label: optionsSelected?.eventCategoryField }]
+        : undefined,
+    [optionsSelected?.eventCategoryField]
+  );
+  const tiebreakerField = useMemo(
+    () =>
+      optionsSelected?.tiebreakerField != null
+        ? [{ label: optionsSelected?.tiebreakerField }]
+        : undefined,
+    [optionsSelected?.tiebreakerField]
+  );
+  const timestampField = useMemo(
+    () =>
+      optionsSelected?.timestampField != null
+        ? [{ label: optionsSelected?.timestampField }]
+        : undefined,
+    [optionsSelected?.timestampField]
+  );
+
+  return (
+    <Container>
+      <FlexGroup alignItems="center" justifyContent="spaceBetween" gutterSize="none">
+        <EuiFlexItem>
+          {errors.length > 0 && (
+            <ErrorsPopover ariaLabel={i18n.EQL_VALIDATION_ERROR_POPOVER_LABEL} errors={errors} />
+          )}
+          {isLoading && <Spinner data-test-subj="eql-validation-loading" size="m" />}
+        </EuiFlexItem>
+        {!onSelectLanguage && (
+          <EuiFlexItem grow={false}>
+            <EqlOverviewLink />
+          </EuiFlexItem>
+        )}
+        {onSelectLanguage && (
+          <>
+            <FlexItemWithMarginRight grow={false}>
+              <EqlOverviewLink />
+            </FlexItemWithMarginRight>
+            <FlexItemLeftBorder grow={false}>
+              <EuiPopover
+                button={
+                  <EuiButtonIcon
+                    onClick={openEqlSettingsHandle}
+                    iconType="controlsVertical"
+                    aria-label="eql settings"
+                  />
+                }
+                isOpen={openEqlSettings}
+                closePopover={closeEqlSettingsHandle}
+                anchorPosition="downCenter"
```

Comment:
The `EQL Settings` popover does not own keyboard focus, per the following gif:

![eql_settings_focus](https://user-images.githubusercontent.com/4459398/107453675-ab226300-6b08-11eb-944e-aae103023d99.gif)

Adding `ownFocus={true}` to the popover solves the focus issue, but the `Event category field` select's field chooser is automatically displayed when `ownFocus={true}` is added, so a bit of experimentation may be required to get the popover to own focus without this unwanted side effect.